daemon/container_unix.go

```diff
@@ -896,8 +900,6 @@ func (container *Container) configureNetwork(networkName, service, networkDriver
 		}
 	}
 
-	container.updateSandboxNetworkSettings(sb)
```

Comment:
@estesp @aboch that's correct. we can undo the removal of `updateSandboxNetworkSettings`.
But, to make things right, we should not set `NamespacePath: c.NetworkSettings.SandboxKey,` in `populateCommand` for native driver since it will not be used after this PR.
Though having it there will not cause any extra harm, it is better to be explicit and not set it when `SupportHooks` is enabled. WDYT ?